lib/salus/scanners/yarn_audit.rb

```diff
@@ -127,6 +133,91 @@ def handle_legacy_yarn_audit
       report_failure
     end
 
+    def auto_fix_vulns
+      @vulns_w_paths.each do |vuln|
+        path = vuln["Path"]
+        package = vuln["Package"]
+        # vuln["Path"] for some indirect dependency "foo"
+        # looks like "foo > bar > baz".
+        # vuln["Path"] for some direct dependency "foo"
+        # looks like "foo"
+        is_direct_dep = path == package
+
+        # TODO: Fix indirect dependencies as well
+        fix_direct_dependency(vuln) if is_direct_dep
+      end
+    end
+
+    def fix_direct_dependency(vuln)
+      package = vuln["Package"]
+      patched_version_range = vuln["Patched in"]
+
+      vulnerable_package_info = run_shell("yarn info #{package} --json").stdout
+      begin
+        vulnerable_package_info = JSON.parse(vulnerable_package_info)
+      rescue StandardError
+        report_error("yarn info #{package} --json did not return JSON")
+        return
+      end
+      list_of_versions = vulnerable_package_info["data"]["versions"]
+      patched_version = select_upgrade_version(patched_version_range, list_of_versions)
+
+      if !patched_version.nil?
+        if !@packages.dig("dependencies", package).nil?
+          @packages["dependencies"][package] = "^#{patched_version}"
+        end
+
+        if !@packages.dig("resolutions", package).nil?
+          @packages["resolutions"][package] = "^#{patched_version}"
+        end
+
+        if !@packages.dig("devDependencies", package).nil?
+          @packages["devDependencies"][package] = "^#{patched_version}"
+        end
+      end
+    end
+
+    def select_upgrade_version(patched_version_range, versions_list)
+      version_range_details =
+        patched_version_range
+          .match(/(?<operator>(<|>)?(=|~)?)?(?<major>\d+)\.(?<minor>\d+)\.(?<patch>\d+)/)
```

Comment:
Can `^` appear as an operator?

Reply:
Also, will yarn always output versions in the right format? like `1.2.3`?
Could it the version be something weird like `1A.2B.C`, or `1234`?

Reply:
We can generally assume versions will follow `1.2.3` as you mentioned, because [NPM recommends semantic versioning](https://docs.npmjs.com/about-semantic-versioning) for packages. There may be edge cases that don't follow the standard, though.